### PR TITLE
Remove Rotate and Resize Actions

### PR DESCRIPTION
**Pull request by tinxx**
This MR removes the now obsolete actions for scaling and rotating a node along with the gizmos.

The resize node action has been improved to allow for resize on the up-axis. Additionally, sprite handles now indicate the resize direction. The handles use a new dedicated shader that honors portals and transparency. The up/down handle is repositioned to the side of the node that faces the camera so that it should be easily visible without the clutter of adding four of them.

screenshot-2024-10-29-09-22-29

Background:
As node dimensions are usually derived from code metrics, rotation and resize are generally not a good idea.
However, there are exceptions that can be flagged to `AllowManualResize`. These are, as of now, the architecture nodes that need to be resized to allow for other nodes being placed on them.

### CODE REVIEW

File: Assets/Resources/Shaders/TransparentSpritePortalShader.shader [hunk 1]
```diff
@@ -0,0 +1,86 @@
+Shader "Unlit/TransparentSpritePortalShader"
+{
+    Properties
+    {
+        _MainTex ("Texture", 2D) = "white" {}
```
Comment (falko17) [inline Assets/Resources/Shaders/TransparentSpritePortalShader.shader:5]: There probably shouldn't be a space here:
```suggestion
        _MainTex("Texture", 2D) = "white" {}
```

File: Assets/Resources/Shaders/TransparentSpritePortalShader.shader [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        // Clipping
+        _PortalMin("Portal Left Front Corner", vector) = (-10, -10, 0, 0)
+        _PortalMax("Portal Right Back Corner", vector) = (10, 10, 0, 0)
+    }
+    SubShader
+    {
+        Tags {
+            "Queue"="Transparent"
+            "RenderType"="Transparent"
+            "IgnoreProjector"="True"
```
Comment (falko17) [inline Assets/Resources/Shaders/TransparentSpritePortalShader.shader:16]: Consistent formatting with the rest here:
```suggestion
            "Queue" = "Transparent"
            "RenderType" = "Transparent"
            "IgnoreProjector" = "True"
```

File: Assets/SEE/Controls/Actions/ActionStateTypes.cs
Comment (falko17) [inline Assets/SEE/Controls/Actions/ActionStateTypes.cs:1]: General comment: It seems that some tests are failing, probably due to the changes in this class. Please investigate and fix this.

File: Assets/SEE/Controls/Actions/ResizeNodeAction.cs [hunk 3]
```diff
@@ -351,6 +375,7 @@ private void Start()
                 }
 
                 this.nodeRef = nodeRef;
+                mainCameraTransform = Camera.main.transform;
```
Comment (falko17) [inline Assets/SEE/Controls/Actions/ResizeNodeAction.cs:378]: I think we already have a cached field for this in `MainCamera.cs`.

File: Assets/SEE/Controls/Actions/ResizeNodeAction.cs [hunk 4]
```diff
@@ -430,17 +459,98 @@ private void InitHandles()
...
+                    if (upDownHandleTransform.localRotation.eulerAngles.y != 0f)
+                    {
+                        upDownHandleTransform.localRotation = Quaternion.Euler(90f, 0f, 0f);
+                        flipScales();
+                    }
+                }
+
+                // The height is updated during the resize process
+                upDownHandleTransform.localPosition = new(
+                        0.5001f * directionToCamera.x,
```
Comment (falko17) [inline Assets/SEE/Controls/Actions/ResizeNodeAction.cs:539]: This `0.5001f` magic number is repeated a few times in this file, can you extract it into a named constant?

File: Assets/SEE/Controls/Actions/ResizeNodeAction.cs [hunk 5]
```diff
@@ -468,184 +578,211 @@ void StartResizing()
...
-                foreach (Transform sibling in parent)
+                // Calculate new scale and position
+                Vector3 cursorMovement;
+                if (currentResizeStep.Up)
                 {
-                    if (sibling != transform && sibling.gameObject.IsNodeAndActiveSelf())
-                    {
-                        siblings.Add(sibling);
-                    }
+                    cursorMovement = new(0f, (currentResizeStep.InitialMousePosition.y - Input.mousePosition.y) / Screen.height, 0f);
```
Comment (falko17) [inline Assets/SEE/Controls/Actions/ResizeNodeAction.cs:591]: Can this method also be called on VR players? If so, we cannot use the mouse position/screen height.

File: Assets/SEE/Controls/Actions/ResizeNodeAction.cs [hunk 6]
```diff
@@ -654,15 +791,26 @@ void UpdateSize()
...
+                                transform.localPosition.x + 0.5f * transform.localScale.x * direction.x,
+                                child.localPosition.y, // don't move handles up or down
+                                transform.localPosition.z + 0.5f * transform.localScale.z * direction.z);
+                    }
+                    else
+                    {
+                        // Adapt other children's position based on changed position and size
+                        bool shift2D = !child.gameObject.IsNodeAndActiveSelf();
+                        child.localPosition = new(
+                                child.localPosition.x + (shift2D ? 0.5f * posDiff.x * currentResizeStep.Direction.x: 0f),
```
Comment (falko17) [inline Assets/SEE/Controls/Actions/ResizeNodeAction.cs:808]:
```suggestion
                                child.localPosition.x + (shift2D ? 0.5f * posDiff.x * currentResizeStep.Direction.x : 0f),
```

File: Assets/SEE/Controls/Actions/ResizeNodeAction.cs [hunk 7]
```diff
@@ -654,15 +791,26 @@ void UpdateSize()
...
+                                child.localPosition.y + posDiff.y, // we always resize height in positiove direction
```
Comment (falko17) [inline Assets/SEE/Controls/Actions/ResizeNodeAction.cs:809]:
```suggestion
                                child.localPosition.y + posDiff.y, // we always resize height in positive direction
```

File: Assets/SEE/Game/City/VisualNodeAttributes.cs [hunk 8]
```diff
@@ -160,10 +160,10 @@ public string DepthMetric
         [Tooltip("Whether the source name will be added to a node.")]
         public bool ShowNames = false;
         /// <summary>
-        /// Whether users may manipulate (e.g., resize) instances of this type of node.
+        /// Defines if node type may be manually resized by users.
         /// </summary>
-        [Tooltip("Whether users may manipulate (e.g., resize) instances of this type of node.")]
-        public bool AllowManualNodeManipulation = false;
+        [Tooltip("May users resize this type of node?.")]
```
Comment (falko17) [inline Assets/SEE/Game/City/VisualNodeAttributes.cs:165]:
```suggestion
        [Tooltip("May users resize this type of node?")]
```

File: Assets/SEE/Game/SceneManipulation/GameNodeMover.cs [hunk 9]
```diff
@@ -8,6 +8,12 @@ namespace SEE.Game.SceneManipulation
     /// </summary>
     public static class GameNodeMover
     {
+        /// <summary>
+        /// Empty space on the y-axis bewteen the top of the target object and the bottom of the child
```
Comment (falko17) [inline Assets/SEE/Game/SceneManipulation/GameNodeMover.cs:12]:
```suggestion
        /// Empty space on the y-axis between the top of the target object and the bottom of the child
```

File: Assets/SEE/GameObjects/Materials.cs [hunk 10]
```diff
@@ -59,11 +60,15 @@ public enum ShaderType
         /// (located in folder Resources).
         /// </summary>
         private const string drawableDashedLineMaterialName = "Materials/DrawableDashedLineMaterial";
+        /// <summary>
+        /// Name of the material for sprites with transparency within a portal (located in Resources folder).
+        /// </summary>
+        private const string spriteName = "Materials/TransparentSpritePortalMaterial";
```
Comment (falko17) [inline Assets/SEE/GameObjects/Materials.cs:66]: Field should ideally be named `spriteMaterialName` (like the others above it).

**Review by falko17 — CHANGES_REQUESTED** [review]
Thanks, it's always nice to get a PR where thousands of lines can be deleted :D
Just a few comments below, but all of them are minor.

File: Assets/SEE/Controls/Actions/ResizeNodeAction.cs (hunk 5 above)
Comment (falko17) [reply Assets/SEE/Controls/Actions/ResizeNodeAction.cs:596]: I've added an issue #806 which tracks this problem, so we can merge this before handling that part.

**Comment by tinxx** [comment]
> Apart from the compile error, this looks good to me now and can be merged! I'll enable auto-merge, so as soon as the pipeline runs through (i.e., the error is fixed) this should be done.

I have removed the VR code paths for rotate as well.  However, I currently cannot test VR at all but this fixes the
compilation errors.

I think we need someone to test VR regularly because most developers won't be able to do that.

**Pull request merged**